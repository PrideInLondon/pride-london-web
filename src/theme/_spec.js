import { screenSizes, breakpoints } from './theme'

it.each`
  screen         | breakpointIndex
  ${'mobile'}    | ${0}
  ${'tablet'}    | ${1}
  ${'desktop'}   | ${2}
  ${'desktopHD'} | ${3}
`('successfully assigns $screen screen size', ({ screen, breakpointIndex }) => {
  expect(screenSizes[screen]).toBe(breakpoints[breakpointIndex])
})
