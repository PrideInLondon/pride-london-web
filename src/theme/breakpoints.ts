export const sm = 375
export const md = 768
export const lg = 1024
export const xl = 1440
export const nav = 1360

export interface Breakpoints {
  sm: string
  md: string
  lg: string
  xl: string
  nav: string
}

export const breakpoints: string[] & Breakpoints = Object.assign(
  [`${sm}px`, `${md}px`, `${lg}px`, `${xl}px`],
  {
    sm: `${sm}px`,
    md: `${md}px`,
    lg: `${lg}px`,
    xl: `${xl}px`,
    nav: `${nav}px`,
  }
)
