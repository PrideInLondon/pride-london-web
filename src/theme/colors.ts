export const colors = {
  black: '#000000',
  darkGrey: '#5B5B5B',
  mediumGrey: '#E1E1E1',
  lightGrey: '#EFEFEF',
  white: '#FFFFFF',
  yellow: '#FFD95E',
  tomato: '#FF5B44',
  pink: '#ED2C6E',
  fuscia: '#FF88F1',
  electricPurple: '#BC01FF',
  darkPurple: '#660C63',
  indigo: '#2D2F7F',
  darkIndigo: '#20215A',
  greyBlue: '#5770D1',
  beachBlue: '#2C77E5',
  skyBlue: '#2ACFFF',
  bondiBlue: '#00A3BD',
  darkCyan: '#008296',
  eucalyptusGreen: '#2CDA9D',
  lemonGreen: '#C3D30E',
  green: '#34C30A',
  amber: '#FFCA00',
  red: '#D0021B',
  gold: '#FFE16C',
  silver: '#E0DFDC',
  bronze: '#B59035',
  mexicanPink: '#E6007E',
  raisinBlack: '#272727',
}
