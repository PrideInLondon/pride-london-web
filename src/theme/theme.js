const theme = {
  colors: {
    black: '#000000',
    darkGrey: '#5B5B5B',
    mediumGrey: '#E1E1E1',
    lightGrey: '#EFEFEF',
    white: '#FFFFFF',
    yellow: '#FFD95E',
    tomato: '#FF5B44',
    pink: '#ED2C6E',
    fuscia: '#FF88F1',
    electricPurple: '#BC01FF',
    darkPurple: '#660C63',
    indigo: '#2D2F7F',
    greyBlue: '#5770D1',
    beachBlue: '#2C77E5',
    skyBlue: '#2ACFFF',
    bondiBlue: '#00A3BD',
    eucalyptusGreen: '#2CDA9D',
    lemonGreen: '#C3D30E',
    green: '#34C30A',
    amber: '#FFCA00',
    red: '#D0021B',
  },
  fonts: {
    title: 'Poppins, sans-serif',
    body: 'Roboto, sans-serif',
  },
  space: [0, 5, 10, 15, 20],
  breakpoints: [375, 768, 1024, 1440],
  navBreakpoint: 1275,
}

export default theme
