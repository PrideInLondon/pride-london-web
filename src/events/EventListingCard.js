import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { colors } from '../theme/colors'
import { fonts } from '../theme/fonts'
import { mediaQueries } from '../theme/mediaQueries'
import {
  Card,
  CardImage,
  CardContent,
  CardTitle,
  CardFooter,
} from '../components/card'
import CalendarIcon from '../components/icons/calendarIcon'
import { generateEventSlug, momentizeRecurrenceDate } from './helpers'
import { generateDisplayDate } from './EventCard'

const CardDate = styled.span`
  display: flex;
  color: ${colors.darkCyan};
  font-size: 0.75rem;
  line-height: 1.29;
  margin-bottom: 0.65rem;
  font-family: ${fonts.title};
  font-weight: 600;
  ${mediaQueries.md} {
    font-size: 0.875rem;
  }
`

const StyledCardFooter = styled(CardFooter)`
  flex-direction: row;
  justify-content: space-between;
`

const Details = styled.span`
  color: ${colors.indigo};
  text-decoration: none;
  font-size: 1rem;
  padding-bottom: 2px;
  background-image: linear-gradient(
    180deg,
    transparent calc(100% - 2px),
    ${colors.darkCyan} 2px
  );
  background-size: 100% 100%;
  ${mediaQueries.md} {
    font-size: 1.125rem;
  }
`

const Price = styled.div`
  align-self: flex-end;
  color: ${colors.darkCyan};
  font-family: Poppins;
  font-weight: 600;
  font-size: 0.875rem;
  line-height: 1.25;
  letter-spacing: -0.16px;

  ${mediaQueries.md} {
    font-size: 1rem;
  }
`

const Location = styled.p`
  color: ${colors.darkIndigo};
  font-size: 0.75rem;
  font-weight: 300;
  line-height: 1.29;
  ${mediaQueries.md} {
    font-size: 0.875rem;
  }
`

const PaddedCalendarIcon = styled(CalendarIcon)`
  margin-right: 8.5px;
  height: 1rem;
`

const When = ({ startTime, endTime, recurrenceDates }) => (
  <CardDate>
    <PaddedCalendarIcon color={colors.darkCyan} />
    {generateDisplayDate({
      start: new Date(startTime),
      end: new Date(endTime),
      lastOccurrence:
        recurrenceDates &&
        new Date(
          momentizeRecurrenceDate(recurrenceDates[recurrenceDates.length - 1])
        ),
    })}
  </CardDate>
)

When.propTypes = {
  startTime: PropTypes.string.isRequired,
  endTime: PropTypes.string.isRequired,
  recurrenceDates: PropTypes.arrayOf(PropTypes.string),
}

When.defaultProps = {
  recurrenceDates: [],
}

export const EventListingCard = ({ event, variant }) => {
  return (
    <Card variant={variant} to={generateEventSlug(event)}>
      <CardImage
        image={event.eventsListPicture.fixed}
        alt={event.eventsListPicture.title}
      />
      <CardContent>
        <When {...event} />
        <CardTitle>{event.name}</CardTitle>
        <Location>
          {event.locationName}, {event.addressLine1}
        </Location>
        <StyledCardFooter>
          <div>
            <Details>Event details</Details>
          </div>
          <Price>
            {event.eventPriceLow === 0
              ? 'Free entry'
              : `From Â£${
                  event.eventPriceLow % 1 === 0
                    ? event.eventPriceLow
                    : event.eventPriceLow.toFixed(2)
                }`}
          </Price>
        </StyledCardFooter>
      </CardContent>
    </Card>
  )
}

EventListingCard.propTypes = {
  event: PropTypes.object.isRequired,
  variant: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.shape({
      default: PropTypes.string,
      sm: PropTypes.string,
      md: PropTypes.string,
      lg: PropTypes.string,
      xl: PropTypes.string,
    }),
  ]),
}

EventListingCard.defaultProps = {
  variant: 'column',
}

export default EventListingCard
