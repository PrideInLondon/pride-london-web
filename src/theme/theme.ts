import { breakpoints } from './breakpoints'
import { mediaQueries } from './mediaQueries'
import { colors } from './colors'

const theme = {
  colors,
  fonts: {
    title: 'Poppins, sans-serif',
    body: 'Roboto, sans-serif',
  },
  space: [0, 5, 10, 15, 20],
  breakpoints,
  mediaQueries,
  navBreakpoint: 1360,
}

export default theme
