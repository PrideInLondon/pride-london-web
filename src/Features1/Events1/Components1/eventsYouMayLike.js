import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import moment from 'moment'
import ChevronRight from '../../../components/chevronRight'
import { Consumer } from '../../../components/appContext'
import { Container, Row, Column } from '../../../components/grid'
import { media } from '../../../theme/media'
import EventListingCard from './eventListingCard'

const ViewAll = styled.a`
  color: ${props => props.theme.colors.indigo};
  font-family: ${props => props.theme.fonts.title};
  font-size: 1rem;
  padding-top: 5px;
  text-align: right;
  text-decoration: none;
`

const FlexColumn = styled(Column)`
  display: flex;
`

export const StyledContainer = styled(Container)`
  padding: 30px 0px;
  ${media.desktop`
    padding: 60px 0px;
  `}
  background-color: ${props => props.theme.colors.lightGrey};
`

const Heading = styled.h1`
  font-size: 1.25rem;
  margin: 0;
  ${media.desktop`
    font-size: 2rem;
  `};
`

const DesktopOnly = styled.span`
  display: none;
  ${media.desktop`
    display: inline;
  `};
`

const HeadingRow = styled(Row)`
  margin-bottom: 20px;
  /* add padding to match column padding */
  padding: 0px 10px;
  ${media.desktop`
    padding: 0px 15px;
  `};
  align-items: center;
  justify-content: space-between;
`

const filterEventsYouMayLike = (events, eventId) => {
  const filteredEvents = events.filter(event => {
    if (event.node.id === eventId) return false

    return moment(event.node.startTime).diff(moment(), 'hours') > 0
  })

  return filteredEvents.splice(0, 3)
}

export const EventsYouMayLike = ({ eventId }) => (
  <Consumer>
    {context => {
      const eventsYouMayLike = filterEventsYouMayLike(
        context.state.events,
        eventId
      )

      if (eventsYouMayLike.length === 0) return null

      return (
        <StyledContainer>
          <HeadingRow>
            <Heading>You may also like</Heading>
            <ViewAll href="/events">
              View all<DesktopOnly>&nbsp;events</DesktopOnly>&nbsp;
              <ChevronRight />
            </ViewAll>
          </HeadingRow>
          <Row>
            {eventsYouMayLike.map(event => (
              <FlexColumn width={[1, 1, 1 / 2, 1 / 3]} key={event.node.id}>
                <EventListingCard event={event.node} />
              </FlexColumn>
            ))}
          </Row>
        </StyledContainer>
      )
    }}
  </Consumer>
)

EventsYouMayLike.propTypes = {
  eventId: PropTypes.string.isRequired,
}

export default EventsYouMayLike
