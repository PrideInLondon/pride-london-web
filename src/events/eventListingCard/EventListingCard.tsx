import React from 'react'
import { colors } from '../../theme/colors'
import {
  Card,
  CardImage,
  CardContent,
  CardTitle,
  CardFooter,
} from '../../components/card'
import { generateEventSlug } from '../helpers'
import { generateDisplayDate, formatLocation } from './EventListingCard.utils'
import { EventListingCardProps, WhenProps } from './EventListingCard.types'
import {
  CardDate,
  Details,
  Location,
  PaddedCalendarIcon,
  Price,
} from './EventListingCard.styles'

const When: React.FC<WhenProps> = ({ dates }) => {
  return (
    <CardDate>
      <PaddedCalendarIcon color={colors.darkCyan} />
      {generateDisplayDate({
        dates,
      })}
    </CardDate>
  )
}

export const EventListingCard: React.FC<EventListingCardProps> = ({
  event: {
    id,
    name,
    eventsListPicture,
    date: { dates },
    location2: platform,
    locationName,
    addressLine1,
    eventPriceLow,
  },
  variant,
}) => (
  <Card variant={variant} to={generateEventSlug({ id, name })}>
    <CardImage image={eventsListPicture.fixed} alt={eventsListPicture.title} />
    <CardContent>
      <When dates={dates} />
      <CardTitle>{name}</CardTitle>
      <Location>
        {formatLocation({
          platform,
          locationName,
          addressLine1,
        })}
      </Location>
      <CardFooter flexDirection="row" justifyContent="space-between">
        <div>
          <Details>Event details</Details>
        </div>
        <Price>
          {eventPriceLow === 0
            ? 'Free entry'
            : `From Â£${
                eventPriceLow % 1 === 0
                  ? eventPriceLow
                  : eventPriceLow.toFixed(2)
              }`}
        </Price>
      </CardFooter>
    </CardContent>
  </Card>
)

EventListingCard.defaultProps = {
  variant: 'column',
}

export default EventListingCard
