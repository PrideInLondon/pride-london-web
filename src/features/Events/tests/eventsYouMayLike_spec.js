import React from 'react'
import { shallow } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import { EventsYouMayLike } from '../eventsYouMayLike'
import EventListingCard from '../eventListingCard'
import theme from '../../../theme/theme'

beforeEach(() => {
  jest.resetModules()
})

jest.mock('../../../components/appContext', () => {
  const nodeOne = {
    node: {
      id: '123',
      startTime: '2035-06-02T13:30+01:00',
      eventsListPicture: {
        file: {
          url: '123123',
        },
      },
    },
  }

  const nodeTwo = {
    node: {
      id: '234',
      startTime: '2006-06-02T13:30+01:00',
      eventsListPicture: {
        file: {
          url: '123123',
        },
      },
    },
  }

  const nodeThree = {
    node: {
      id: '1234',
      startTime: '2035-06-02T13:30+01:00',
      eventsListPicture: {
        file: {
          url: '123123',
        },
      },
    },
  }

  return {
    Consumer: props =>
      props.children({
        state: {
          events: [nodeOne, nodeTwo, nodeThree],
        },
      }),
  }
})

const shallowWithTheme = tree => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()
  return shallow(tree, { context })
}

describe('The Events You May Like component', () => {
  const mockProps = {
    eventId: '1234',
  }

  it('should matchsnapshot', () => {
    const wrapper = shallowWithTheme(<EventsYouMayLike {...mockProps} />)

    expect(wrapper.dive()).toMatchSnapshot()
  })

  it('should render Event Listing Card if a correct event is passed from the state', () => {
    const wrapper = shallowWithTheme(<EventsYouMayLike {...mockProps} />)

    expect(wrapper.dive().find(EventListingCard).length).toBe(1)
  })

  it('should not render an item that started in the past', () => {
    const wrapper = shallowWithTheme(<EventsYouMayLike {...mockProps} />)

    expect(
      wrapper
        .dive()
        .find(EventListingCard)
        .props().event.startTime
    ).not.toBe('2006-06-02T13:30+01:00')
  })

  it('should not render an item that has the same id as the current event', () => {
    const wrapper = shallowWithTheme(<EventsYouMayLike {...mockProps} />)

    expect(
      wrapper
        .dive()
        .find(EventListingCard)
        .props().event.id
    ).not.toBe(mockProps.eventId)
  })
})
