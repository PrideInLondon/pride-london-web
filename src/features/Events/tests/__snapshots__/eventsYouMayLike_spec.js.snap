// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Events You May Like component should matchsnapshot 1`] = `
<Styled(Styled(Box))>
  <Styled(Row)>
    <styled.h1>
      You may also like
    </styled.h1>
    <styled.a
      href="/events"
    >
      View all
      <styled.span>
         events
      </styled.span>
       
      <ChevronRight />
    </styled.a>
  </Styled(Row)>
  <Row>
    <Styled(Column)
      key="123"
      width={
        Array [
          1,
          1,
          0.5,
          0.3333333333333333,
        ]
      }
    >
      <EventListingCard
        event={
          Object {
            "eventsListPicture": Object {
              "file": Object {
                "url": "123123",
              },
            },
            "id": "123",
            "startTime": "2035-06-02T13:30+01:00",
          }
        }
      />
    </Styled(Column)>
  </Row>
</Styled(Styled(Box))>
`;
